src/citrine/resources/file_link.py

```diff
@@ -357,6 +372,81 @@ def get(self,
+        """
+        Get an element of the collection by its id.
+
+        Parameters
+        ----------
+        uid: Union[UUID, str]
+            A representation of the FileLink (Citrine id or file name)
+        version: Optional[UUID, str, int]
+            The version, as a UUID or str(UUID) of the version_id or an int or
+            str(int) of the version number.  If None, returns the file with the
+            highest version number (most recent).
+
+        Returns
+        -------
+        ResourceType
+            An object with specified scope and uid
+
+        """
+        if not isinstance(uid, (str, UUID)):
+            raise TypeError(f"File Link can only be resolved from str or UUID."
+                            f"Instead got {type(uid)} {uid}.")
+        if version is not None and not isinstance(version, (str, UUID, int)):
+            raise TypeError(f"Version can only be resolved from str, int or UUID."
+                            f"Instead got {type(uid)} {uid}.")
+
+        try:  # Check if the uid string is actually a UUID
+            if isinstance(uid, str):
+                uid = UUID(uid)
+        except ValueError:
+            pass
+
+        try:  # Check if the version string is actually a UUID
+            if isinstance(version, str):
+                version = UUID(version)
+        except ValueError:
+            pass
+
+        try:  # Check if the version string is actually an int / version number
+            if isinstance(version, str):
+                version = int(version)
+        except ValueError:
+            pass
+
+        if isinstance(version, str):
+            raise ValueError(
+                f"Version {version} could not be converted to either an int or a UUID"
+            )
+
+        if isinstance(uid, str):
+            # Assume it's the filename on platform;
+            file = None
+            if version is None or isinstance(version, int):
+                file = self._search_by_file_name(dset_id=self.dataset_id, file_name=uid, file_version_number=version)
+            elif isinstance(version, UUID):
+                file = self._search_by_file_version_id(dset_id=self.dataset_id, file_version_id=version)
+            if file is None:
+                raise NotFound(f"Found no file named {uid}")
+            else:
+                return file
+        elif isinstance(uid, UUID):
+            file = None
+            if version is None or isinstance(version, int):
+                file = self._search_by_dataset_file_id(dataset_file_id=uid, dset_id=self.dataset_id,
+                                                       file_version_number=version)
+            elif isinstance(version, UUID):
+                file = self._search_by_file_version_id(dset_id=self.dataset_id, file_version_id=version)
+            if file is None:
+                raise NotFound(f"Found no file named {uid}")
+            else:
+                return file
```

Comment:
I believe it's more pythonic to guarantee that the file is initialized along some branch of a conditional, throwing an exception if not.  Something like (since exception is thrown by checked_request):

```suggestion
        if isinstance(uid, str):
            # Assume it's the filename on platform;
            if version is None or isinstance(version, int):
                file = self._search_by_file_version_id(dset_id=self.dataset_id, file_version_id=version)
            else:  # We did our type checks earlier; version is an int or None
                file = self._search_by_file_name(dset_id=self.dataset_id, file_name=uid, file_version_number=version)
        else:  # We did our type checks earlier; uid is a UUID
            if isinstance(version, UUID):
                file = self._search_by_file_version_id(dset_id=self.dataset_id, file_version_id=version)
            else:  # We did our type checks earlier; version is an int or None
                file = self._search_by_dataset_file_id(dataset_file_id=uid, dset_id=self.dataset_id,
                                                       file_version_number=version)

        return file
```

You could do a catch & release if you wanted to change the exception error:
```
        try:
            if isinstance(uid, str):
            ...
        except NotFound:
            raise NotFound(f"Found no file named {uid}")